addons/mail/models/mail_channel.py

```diff
@@ -281,6 +281,7 @@ def action_unfollow(self):
     def _action_unfollow(self, partner):
         channel_info = self.channel_info('unsubscribe')[0]  # must be computed before leaving the channel (access rights)
         result = self.write({'channel_partner_ids': [(3, partner.id)]})
+        channel_info['is_pinned'] = False
```

Comment:
I added this comment on my PR, maybe look to add something similar here ` # side effect of unsubscribe that wasn't taken into account because channel_info was called before`